src/pudl_scrapers/spiders/ferc2.py

```diff
@@ -0,0 +1,114 @@
+"""Scrapy spider for downloading historical FERC Form 2 Visual FoxPro data."""
+from pathlib import Path
+
+import scrapy
+from scrapy.http import Request
+
+from pudl_scrapers import items
+from pudl_scrapers.helpers import new_output_dir
+
+
+class Ferc2Spider(scrapy.Spider):
+    """Scrapy spider for downloading historical FERC Form 2 Visual FoxPro data."""
+
+    name = "ferc2"
+    allowed_domains = ["www.ferc.gov"]
+    start_urls = [
+        "https://www.ferc.gov/industries-data/natural-gas/industry-forms/form-2-2a-3-q-gas-historical-vfp-data"
+    ]
+
+    def start_requests(self):
+        """Start requesting FERC 2 forms.
+
+        Yields:
+            List of Requests for FERC 2 forms
+        """
+        # Spider settings are not available during __init__, so finalizing here
+        settings_output_dir = Path(self.settings.get("OUTPUT_DIR"))
+        output_root = settings_output_dir / "ferc2"
+        self.output_dir = new_output_dir(output_root)
+
+        yield from self.all_form_requests()
+
+    def parse(self, response):
+        """Produce the FERC 2 item.
+
+        Args:
+            response: scrapy.http.Response containing FERC Form 2 data
+
+        Yields:
+            FERC Form 2 item
+        """
+        if response.meta["part"] is None:
+            path = self.output_dir / f"ferc2-{response.meta['year']}.zip"
+        else:
+            path = (
+                self.output_dir
+                / f"ferc2-{response.meta['year']}-{response.meta['part']}.zip"
+            )
+
+        yield items.Ferc2(
+            data=response.body,
+            year=response.meta["year"],
+            part=response.meta["part"],
+            save_path=path,
+        )
+
+    def form_for_year_part(self, year: int, part: int | None = None):
+        """Produce a form request for the given year.
+
+        Args:
+            year: Reporting year of the data to be scraped.
+            part: In earlier years data is split into two parts (A-M, and N-Z) which
+                must be downloaded separately. Part indicates which one to get.
+
+        Returns:
+            Request for the FERC 2 form
+
+        """
+        early_urls: dict[tuple(int, int), str] = {
```

Comment:
Definitely a bigger discussion than this review, but I've been thinking when we combine the archiver/scraper repos it might be a good time to consolidate all the information about what data is in an archive/how it's organized (partitions, formats, etc.) into a single place. It seems flaky to make PUDL and the scraper/archiver need to understand this separately, and the archiver repo already depends on PUDL, so it should be possible. I think it might make a lot of sense for the Datastore to be responsible for creating, and reading from archives.

Reply:
Yeah, the cross-repo metadata dependencies are not good, since we have multiple tools that depend on the same information. Ideally the scrapers, archivers, and datastore should **not** depend on PUDL. Really nothing should depend on PUDL, as it's more of an end-use application, not a library.

If portions of the code and data stored in the PUDL repository are needed by other tools, it should probably be pulled out of the PUDL repo. Having the data-source specific information be part of the Datastore, and having the datastore be a separate repository, does seem like a reasonable solution.

I'm less clear on if/how the scraping / archiving infrastructure should be linked to the Datastore -- all in one place, or dependent on the Datastore?